solrsearch-plugins/src/test/java/co.cask.hydrator.plugin/SolrSearchSinkTest.java

```diff
@@ -0,0 +1,535 @@
+  }
+
+  @Ignore
+  public void testBatchSolrSearchSink() throws Exception {
+    client = new HttpSolrClient("http://localhost:8983/solr/collection1");
+    String inputDatasetName = "solr-batch-input-source";
+    ETLStage source = new ETLStage("source", MockSource.getPlugin(inputDatasetName));
+
+    Map<String, String> sinkConfigproperties = new ImmutableMap.Builder<String, String>()
+      .put(Constants.Reference.REFERENCE_NAME, "BatchSolrSink")
+      .put("solrMode", SolrSearchSinkConfig.SINGLE_NODE_MODE)
+      .put("solrHost", "localhost:8983")
+      .put("collectionName", "collection1")
+      .put("keyField", "id")
+      .put("batchSize", "1000")
+      .put("outputFieldMappings", "office address:address")
+      .build();
+
+    ETLStage sink = new ETLStage("SolrSink", new ETLPlugin("SolrSearch", BatchSink.PLUGIN_TYPE, sinkConfigproperties,
+                                                           null));
+
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(source)
+      .addStage(sink)
+      .addConnection(source.getName(), sink.getName())
+      .build();
+
+    ApplicationId appId = NamespaceId.DEFAULT.app("testBatchSolrSink");
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(ETLBATCH_ARTIFACT, etlConfig);
+    ApplicationManager appManager = deployApplication(appId.toId(), appRequest);
+
+    DataSetManager<Table> inputManager = getDataset(inputDatasetName);
+    List<StructuredRecord> input = ImmutableList.of(
+      StructuredRecord.builder(inputSchema).set("id", "1").set("firstname", "Brett").set("lastname", "Lee").set
+        ("office address", "NE lake side").set("pincode", 480001).build(),
+      StructuredRecord.builder(inputSchema).set("id", "2").set("firstname", "John").set("lastname", "Ray").set
+        ("office address", "SE lake side").set("pincode", 480002).build()
+    );
+    MockSource.writeInput(inputManager, input);
+
+    MapReduceManager mrManager = appManager.getMapReduceManager(ETLMapReduce.NAME);
+    mrManager.start();
+    mrManager.waitForFinish(5, TimeUnit.MINUTES);
+
+    QueryResponse queryResponse = client.query(new SolrQuery("*:*"));
+    SolrDocumentList resultList = queryResponse.getResults();
+
+    Assert.assertEquals(2, resultList.size());
+    for (SolrDocument document : resultList) {
+      if (document.get("id").equals("1")) {
+        Assert.assertEquals("Brett", document.get("firstname"));
+        Assert.assertEquals("Lee", document.get("lastname"));
+        Assert.assertEquals("NE lake side", document.get("address"));
+        Assert.assertEquals(480001, document.get("pincode"));
+      } else {
+        Assert.assertEquals("John", document.get("firstname"));
+        Assert.assertEquals("Ray", document.get("lastname"));
+        Assert.assertEquals("SE lake side", document.get("address"));
+        Assert.assertEquals(480002, document.get("pincode"));
+      }
+    }
+    // Clean the indexes
+    client.deleteByQuery("*:*");
```

Comment:
Not every test has a call to delete the indexes. Also, Solr client that calls delete command is different for SolrCloud and Single Node mode. Therefore calling delete in test cases itself and not in @After method.